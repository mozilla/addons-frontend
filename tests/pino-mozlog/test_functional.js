const util = require('util');
const exec = util.promisify(require('child_process').exec);

// eslint-disable-next-line amo/describe-with-filename
describe('functional tests', () => {
  it('converts pino logs to mozlog', async () => {
    const { stdout } = await exec(
      'cat tests/pino-mozlog/fixtures/frontend.log | node bin/pino-mozlog.js',
    );

    // eslint-disable-next-line no-console
    console.log(`${stdout}`);
    expect(stdout).toMatchSnapshot();
  });
});
