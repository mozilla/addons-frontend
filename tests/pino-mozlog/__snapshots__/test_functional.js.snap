// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functional tests converts pino logs to mozlog 1`] = `
"{"EnvVersion":"2.0","Fields":{"msg":"Sentry reporting is disabled; Set config.sentryDsn to enable it."},"Hostname":"big","Logger":"amo","Pid":85547,"Severity":4,"Timestamp":1612950421902131000,"Type":"app.log","severity":400}
{"EnvVersion":"2.0","Fields":{"msg":"üî• Addons-frontend server is running [ENV:development] [isDevelopment:true] [isDeployed:false] [apiHost:http://localhost:3000] [apiPath:/api/] [apiVersion:v5]"},"Hostname":"big","Logger":"amo","Pid":85547,"Severity":6,"Timestamp":1612950421911287000,"Type":"app.log","severity":200}
{"EnvVersion":"2.0","Fields":{"msg":"üö¶ Proxy detected, frontend running at http://127.0.0.1:3333."},"Hostname":"big","Logger":"amo","Pid":85547,"Severity":7,"Timestamp":1612950421911469000,"Type":"app.log","severity":100}
{"EnvVersion":"2.0","Fields":{"msg":"üëÅ  Open your browser at http://127.0.0.1:3000 to view it."},"Hostname":"big","Logger":"amo","Pid":85547,"Severity":7,"Timestamp":1612950421911601000,"Type":"app.log","severity":100}
"
`;
